### PR TITLE
#164139642 Implement user registration on the app.

### PR DESCRIPTION
**Pull request by kleva-j**
#### What does this PR do?
It consumes the user registration endpoint using react and redux.

#### Description of Task to be completed?
Users should be able to signup into Authors Haven by providing a name, username, email and password. On successful registration, the user is required to verify their account using the link sent to their email.

#### How should this be manually tested?
- Visit the app URL `https://vidar-ah-frontend-staging.herokuapp.com/signup`
- Provide your registration details ( name, username, email and password).
- If the credentials are valid, you'll be successfully registered to the app
- If the credentials are invalid, you'll get error(s) messages.
- After registration, verify the account using verification link from the email.
- The link redirects to a page that verifies the user's email.

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/164139642

#### Questions:
N/A

### CODE REVIEW

**Review by youngestdj — DISMISSED** [review]

**Comment by 9jaswag** [comment]
@kleva-j why is this not ready yet

**Review by youngestdj — DISMISSED** [review]

File: src/__tests__/redux/actions/__snapshots__/signup.js.snap [hunk 1]
```diff
@@ -0,0 +1,37 @@
+// Jest Snapshot v1, https://goo.gl/fbAQLP
+
```
Comment (codinger41) [inline src/__tests__/redux/actions/__snapshots__/signup.js.snap]: @kleva-j  I think this file should be ignored

**Review by codinger41 — COMMENTED** [review]
Let's delete the Header.jsx and leave Index.jsx in that header folder and just make sure we use `isLoggedIn` instead of `isSignedUp`

**Review by codinger41 — DISMISSED** [review]

**Review by youngestdj — DISMISSED** [review]

File: src/__tests__/components/header.spec.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { Header } from '../../components/header/Index';
+
+describe('Test Header component', () => {
```
Comment (9jaswag) [inline src/__tests__/components/header.spec.js]: @kleva-j there's no need to add the word `Test` here and every other place it is present.

File: src/__tests__/components/verifyEmail.spec.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  },
+  history: {
+    push: jest.fn(),
+    location: {
+      pathname: '/verify_email'
+    }
+  },
+};
+
+describe('Header shallow tests', () => {
```
Comment (9jaswag) [inline src/__tests__/components/verifyEmail.spec.js]: @kleva-j same here

File: src/__tests__/redux/actions/signup.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+const mockStore = configureMockStore([thunk]);
+
+const mockData = {
+  name: 'vidar-name',
+  username: 'vidar-username',
+  email: 'vidar-email',
+};
+
+describe('<SIGNUP ACTION CREATORS />', () => {
```
Comment (9jaswag) [inline src/__tests__/redux/actions/signup.js]: @kleva-j why is this describe written this way?

File: src/__tests__/redux/actions/signup.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  let store;
+
+  describe('SIGNUP', () => {
```
Comment (9jaswag) [inline src/__tests__/redux/actions/signup.js]: @kleva-j why all caps?

**Review by 9jaswag — CHANGES_REQUESTED** [review]
I have not seen a screen share for this. Please attend to the issues raised

File: src/__tests__/components/header.spec.js (hunk 2 above)
Comment (kleva-j) [reply src/__tests__/components/header.spec.js]: Working on it.

File: src/__tests__/components/verifyEmail.spec.js (hunk 3 above)
Comment (kleva-j) [reply src/__tests__/components/verifyEmail.spec.js]: Now working on this.

File: src/__tests__/redux/actions/signup.js (hunk 4 above)
Comment (kleva-j) [reply src/__tests__/redux/actions/signup.js]: Now working on this.

File: src/__tests__/redux/actions/signup.js (hunk 5 above)
Comment (kleva-j) [reply src/__tests__/redux/actions/signup.js]: Now working on this.

**Review by olajide1234 — DISMISSED** [review]

**Review by youngestdj — DISMISSED** [review]

**Review by olajide1234 — DISMISSED** [review]

**Comment by 9jaswag** [comment]
@kleva-j @leksyib @youngestdj @olajide1234 @Dubby20 @chingsley why is this still waiting for peer review 13 days later?

**Review by youngestdj — DISMISSED** [review]

**Comment by 9jaswag** [comment]
did Y'all actually review this or just gave it a pass to fulfill all righteousness?

**Comment by youngestdj** [comment]
@9jaswag I already reviewed it, he only made minor changes.

**Comment by 9jaswag** [comment]
@kleva-j @olajide1234 why are you guys adding the ready to merge label?

**Review by olajide1234 — DISMISSED** [review]

**Review by youngestdj — DISMISSED** [review]

**Review by olajide1234 — APPROVED** [review]

**Review by youngestdj — APPROVED** [review]

**Review by 9jaswag — APPROVED** [review]

**Pull request merged**